python/lsst/summit/utils/blockUtils.py

```diff
@@ -0,0 +1,438 @@
+            specified, all data for the specified block is returned.
+
+        Returns
+        -------
+        data : `pandas.DataFrame`
+            The row data.
+        """
+        rowsForBlock = self.data[self.data['blockNum'] == block]
+        if seqNum is None:
+            return rowsForBlock
+        return rowsForBlock[rowsForBlock['blockSeqNum'] == seqNum]
+
+    def printBlockEvolution(self, block, seqNum=None):
+        """Display the evolution of the specified block.
+
+        If the seqNum is specified, the evolution of that specific block
+        exection is displayed, otherwise all executions of that block are
+        printed.
+
+        Parameters
+        ----------
+        block : `int`
+            The block number to get the events for.
+        seqNum : `int`, optional
+            The sequence number, if specified, to print the evolution of. If
+            not specified, all sequence numbers for the block are printed.
+        """
+        if seqNum is None:
+            seqNums = self.getSeqNums(block)
+        else:
+            seqNums = [seqNum]
+        print(f'Evolution of BLOCK {block} for dayObs={self.dayObs} {seqNum=}:')
+        for seqNum in seqNums:
+            blockInfo = self.getBlockInfo(block, seqNum)
+            print(blockInfo, '\n')
+
+    def getBlockInfo(self, block, seqNum):
+        """Get the block info for the specified block.
+
+        Parses the rows relating to this block execution, and returns
+        the information as a ``BlockInfo`` instance.
+
+        Parameters
+        ----------
+        block : `int`
+            The block number.
+        seqNum : `int`
+            The sequence number.
+
+        Returns
+        -------
+        blockInfo : `lsst.summit.utils.blockUtils.BlockInfo`
+            The block info.
+        """
+        rows = self.getRows(block, seqNum=seqNum)
+        if rows.empty:
+            print(f'No {seqNum=} on dayObs={self.dayObs} for {block=}')
+            return
+
+        blockIds = set()
+        tickets = set()
+        salIndices = set()
+        statePoints = []
+        sitcomPattern = r"SITCOM-(\d+)"
+
+        for index, row in rows.iterrows():
+            salIndices.add(row['salIndex'])
+            blockIds.add(row['blockId'])
+
+            lastCheckpoint = row['lastCheckpoint']
+            sitcomMatches = re.findall(sitcomPattern, lastCheckpoint)
+            tickets.update(sitcomMatches)
+
+            time = efdTimestampToAstropy(row['private_efdStamp'])
+            state = ScriptState(row['state'])
+            reason = row['reason']
+            statePoint = ScriptStatePoint(time=time, state=state, reason=reason)
+            statePoints.append(statePoint)
+
+        # likewise for the blockIds
+        if len(blockIds) > 1:
+            raise RuntimeError(f"Found multiple blockIds ({blockIds}) for {seqNum=}")
+        blockId = blockIds.pop()
+
+        blockInfo = BlockInfo(
```

Comment:
Do you need to preserve the relation between specific `salIndices` and `tickets` from the same row of the block? If so, that has been lost here.

Reply:
I am almost certainly that I don't, no, but that's a good point, thanks!